File: Ruby/ruby/RPG/ninja.rb
require_relative "human" 

class Ninja < Human 
    def initialize
        @stealth = 175 
    end 
    
    def steal(target)
        target.health -= 10
        @health += 10 
        self 
    end 

    def get_away
        @health -= 15
    end 
end 

File: Java/Spring/ObjectMaster/src/com/objectmaster/models/Ninja.java
package com.objectmaster.models;

public class Ninja extends Human {

	public Ninja() {
		this.stealth = 10;
	}
	
	public void steal(Human target) {
		int damage = this.stealth;
		target.health -= damage;
		this.health += damage;
	}
	
	public void runAway() {
		this.health -= 10;
	}
}
